apriltag_ros/src/continuous_detector.cpp

```diff
@@ -221,6 +234,14 @@ void ContinuousDetector::imageCallback (
   // Publish detected tags in the image by AprilTag 2
   tag_detections_publisher_.publish(
       tag_detector_->detectTags(cv_image_,camera_info));
+  
+  // only reset the timer if the validations were passed
+  if (!detector_validations_failed)
```

Comment:
Do we really need a `ros::Timer` for this?

IMHO it'd be better to have a `ros::Time last_valid_detection_;` attribute that you set to the the `header.stamp` when the detection passes the checks, and then we check `if ((header.stamp - last_valid_detection_).toSec() > detection_timeout_)` where appropriate.

Reply:
Yea I could do it like that too (I agree that it's cleaner), will change